libs/sdk-ui/src/execution/useExecution.ts

```diff
@@ -12,27 +18,32 @@ export interface IUseExecutionConfig {
      * Data series will be built using the provided measures that are optionally further scoped for
      * elements of the specified attributes.
      */
-    seriesBy: IAttributeOrMeasure[];
+    seriesBy: ValuesOrPlaceholders<IAttribute | AnyMeasure>;
 
     /**
      * Optionally slice all data series by elements of these attributes.
      */
-    slicesBy?: IAttribute[];
+    slicesBy?: ValuesOrPlaceholders<IAttribute>;
 
     /**
      * Optionally include these totals among the data slices.
      */
-    totals?: ITotal[];
+    totals?: ValuesOrPlaceholders<ITotal>;
 
     /**
      * Optional filters to apply on server side.
      */
-    filters?: INullableFilter[];
+    filters?: ValuesOrPlaceholders<INullableFilter>;
 
     /**
      * Optional sorting to apply on server side.
      */
-    sortBy?: ISortItem[];
+    sortBy?: ValuesOrPlaceholders<ISortItem>;
+
+    /**
+     * Optional resolution context for composed placeholders.
+     */
+    placeholdersResolutionContext?: any;
```

Comment:
these definitions are copy-pasted in multiple places. no need to fix now, it was existing problem. please create followup under FET-512 to deduplicate.